test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -1177,35 +1311,95 @@ void ProtoConverter::closeBlockScope()
 		);
 		updateFunctionMaps(f);
 	}
+	// Pop back the vector of scoped functions.
 	if (!m_scopeFuncs.empty())
 		m_scopeFuncs.pop_back();
-	if (!m_inFunctionDef)
+
+	// If block belongs to function body, then remove
+	// local variables in function body that are going out of scope.
+	if (m_inFunctionDef)
 	{
-		if (!m_globalVars.empty())
-			m_globalVars.pop_back();
+		yulAssert(!m_funcVars.empty(), "Proto fuzzer: Invalid data structure");
+		if (!m_funcVars.back().empty())
+			m_funcVars.back().pop_back();
 	}
+	// Remove variables declared in vanilla block from current
+	// global scope.
 	else
 	{
-		// Variables that have been declared in a
-		// function block, go out of scope
-		if (!m_funcVars.empty())
-			if (!m_funcVars.back().empty())
-				m_funcVars.back().pop_back();
+		yulAssert(!m_globalVars.empty(), "Proto fuzzer: Invalid data structure");
+		m_globalVars.pop_back();
 	}
 }
 
 void ProtoConverter::closeFunctionScope()
 {
-	if (!m_funcVars.empty())
-		m_funcVars.pop_back();
+	yulAssert(!m_funcVars.empty(), "Proto fuzzer: Invalid data structure");
+	m_funcVars.pop_back();
+	yulAssert(!m_funcForLoopInitVars.empty(), "Proto fuzzer: Invalid data structure");
+	m_funcForLoopInitVars.pop_back();
 }
 
 void ProtoConverter::addVarsToScope(vector<string> const& _vars)
 {
+	// If we are in function definition, add the new vars to current function scope
 	if (m_inFunctionDef)
-		m_funcVars.back().back().insert(m_funcVars.back().back().end(), _vars.begin(), _vars.end());
+	{
+		// If we are directly in for-init block, add the newly created vars to the
+		// stack of for-init variables.
+		if (m_inForInitScope && m_forInitScopeExtEnabled)
+		{
+			yulAssert(
+				!m_funcForLoopInitVars.empty() && !m_funcForLoopInitVars.back().empty(),
+				"Proto fuzzer: Invalid data structure"
+			);
+			m_funcForLoopInitVars.back().back().insert(
+				m_funcForLoopInitVars.back().back().end(),
+				_vars.begin(),
+				_vars.end()
+			);
+		}
```

Comment:
Change here is accounting for newly minted variable declarations that are generated by multi declaration or assignment statements inside for init block. The function `addVarsToScope` is used to add such vars to a given scope.

Example:

```for { let a, b, c := func_with_three_return_vals() } 1 {} {}```

This code adds variables `a`, `b`, and `c` to the currently visited for-init block in function scope.